### ISSUE TITLE
no-unnecessary-generics shouldn't disallow single-use inner types

### ISSUE THREAD

**Issue by 43081j** [Microsoft/dtslint#223]
Example:

```ts
someFunc<T extends any[]>(arg: SomeType<T>): void;
```

This can't be achieved any other way as far as i know (or if it can be, it'll be some roundabout way).

We probably shouldn't disallow it.
